docs/download-upload-clone/storage/high-level-design.md

```diff
@@ -0,0 +1,163 @@
+# Storage for MarxanCloud project archives - High-level design
+
+Given the general points outlined in the relevant [Brief document](./brief.md),
+this document highlights the high-level design of the storage flow for
+MarxanCloud project archives.
+
+## Key principles
+
+- Always be streaming
+
+As noted in the brief document, apart from small metadata pieces, core data will
+typically be expected to take up in the range of gigabytes of storage space,
+hence the need to efficiently stream these pieces, whether from db to blob
+storage, from blob storage to any post-processing stage and back to blob
+storage, from blob storage to user devices, etc.
+
+- Uncompressed data
+
+Whereas compressing pieces (e.g. via zip) may lead in most cases to significant
+space savings, this does typically introduce computationally intensive steps
+(compression/decompression) as well as making handling pieces, individually and
+as sets/subsets, potentially more cumbersome.
+
+Given the expected use cases (allowing users to share a project archive to be
+then uploaded by themselves or others as a new project, either directly from
+cloud storage or by first downloading it to their local computers), file size
+savings may be important only to some extent: typically when users want to store
+archives locally/offline; when sharing archives directly through their cloud
+storage locations, file size savings may be practically irrelevant as blob
+storage is cheap as well as - at the expected scale - data ingress and egress.
+
+Preference should be given, instead, to file formats (such as GIS file formats)
+that could be streamed while supporting native compression or other strategies
+to limit file sizes to a minimum.
```

Comment:
The file format/compression solution is not clear to me. I did some googling on the subject, and there seems to be a way to create a zip file without actually using compression, and I assume that's what you're going for here, but it's an assumption, so perhaps you could make it clearer as to how exactly this would work.

Another option/suggestion I have is to evaluate if there is any low compression rate, low cost compression algorithm we can use to get some compression for cheap. Optionally, we may even give the user the choice of using compression or not.

Reply:
sure, I can clarify this in the text

the gist is, for project exports that don't need to be served to users (so only used for cloning without the intermediate artifact downloaded to a user's computer), compressing files shouldn't change much in the big scheme of things: yes, storing GeoJSON uncompressed is a huge waste of space, but disk and bandwidth are cheap, especially if there's no actual billable egress from a data centre but everything goes through a local VPC (again, for plain cloning, not when users download an archive artifact), and we can simply get the raw data and start processing it as it comes in.

I'd keep these two considerations separate - on one hand, combining all the pieces in a single file, on the other, whether to compress either single pieces, or a user-downloadable artifact, or both. I'll try to articulate this better in the brief.